sui-execution/latest/sui-adapter/src/temporary_store.rs

```diff
@@ -802,12 +807,23 @@ impl<'backing> TemporaryStore<'backing> {
             let new_storage_rebate =
                 gas_charger.track_storage_mutation(new_object_size, old_storage_rebate);
             object.storage_rebate = new_storage_rebate;
+
```

Comment:
this is a bit intrusive, even though I do not have a good story for where it could go